docs/tutorials/using_pyrokinetics_on_hpc_systems.rst

```diff
@@ -0,0 +1,395 @@
+   if __name__ == "__main__":
+       main()
+
+This can then be run with the batch script ``~/scratch/pyro_job/pyro_job.sh``:
+
+.. code-block:: bash
+
+   #!/bin/bash
+
+   # Slurm settings
+   # --------------
+
+   #SBATCH --job-name=pyro_job      # Job name
+   #SBATCH --mail-user=me@email.com # Where to send mail reporting on the job
+   #SBATCH --mail-type=END,FAIL     # Mail events (NONE, BEGIN, END, FAIL, ALL)
+   #SBATCH --nodes=1                # Number of compute nodes to run on
+   #SBATCH --ntasks-per-node=24     # Number of MPI processes to spawn per node
+   #SBATCH --cpus-per-task=1        # Number of CPUs per MPI process
+   #SBATCH --mem-per-cpu=8gb        # Memory allocated to each CPU
+   #SBATCH --time=00:30:00          # Total time limit hrs:min:sec
+   #SBATCH --output=%x_%j.log       # Log file for stdout outputs
+   #SBATCH --error=%x_%j.err        # Log file for stderr outputs
+   #SBATCH --partition=nodes
+   #SBATCH --account=myaccountname
+
+   # User settings
+   # -------------
+
+   module purge
+   module load Python OpenMPI netCDF-Fortran FFTW.MPI/3.3.10-gompi-2023a
+
+   source $HOME/.local/venv/pyrokinetics/bin/activate
+
+   # Perform runs
+   # ------------
+
+   gs2=$HOME/gs2/bin/gs2
+   pyfile=$HOME/scratch/pyro_job/pyro_job.py
+   template=$HOME/scratch/pyro_job/pyrokinetics_template.in
+
+   python $pyfile $gs2 $template $HOME/scratch/pyro_job/*.geqdsk --psi 0.5
+
+Note that we don't call ``srun`` explicitly here. In order to make these scripts more
+suitable for automation, we haven't specified the names of the equilibrium files we wish
+to process explicitly, and we've made use of full path names.
+
+QCG-PilotJob is highly configurable, and can adapt to much more complex problems than
+this. For example, rather than choosing exactly 8 cores for each job, we may instead set
+a minimum and maximum core count for each job and let QCG-PilotJob manage the actual
+allocations. It also contains features for restarting runs in case we time out.
+
+
+.. _sec-dispatching-hpc:
+
+Dispatching jobs remotely
+-------------------------
+
+To further assist with the automation of our HPC jobs, we can use a tool such as
+`HPC Rocket <https://svenmarcus.github.io/hpc-rocket/>`_ to dispatch jobs to remote HPC
+machines without the need to manually SSH in. This tool is well suited for use in
+Continuous Integration (CI) pipelines, and can also be run from the command line.
+
+HPC Rocket can be easily installed on the user's machine using:
+
+.. code-block:: bash
+
+   $ pip install hpc-rocket
+
+Before using this, note that we'll still need to SSH into the HPC machine in order to
+set up a Python environment and install all the software we'll need.
+
+We can set up a remote job run by creating a new directory, adding the scripts in
+:ref:`sec-advanced-hpc` and any equilibrium files we want, and then adding the file
+``pyro_job.yaml``:
+
+.. code-block:: yaml
+
+   host: hpc.com
+   user: username
+   password: $PASSWORD
```

Comment:
This feels a little dubious (and presumably won't work with systems requiring MFA?) -- might it be worth documenting how to set up a ControlMaster config (e.g. https://support.york.ac.uk/s/article/How-to-create-a-ControlMaster-SSH-session-so-that-you-only-need-to-authenticate-once-for-multiple-SSH-sessions) so that users just authenticate once and then hpc-rocket can reuse this?